dashboard/app/controllers/javabuilder_sessions_controller.rb

```diff
@@ -13,40 +13,105 @@ class JavabuilderSessionsController < ApplicationController
 
   # GET /javabuilder/access_token
   def get_access_token
-    teacher_list = get_teacher_list.join(',')
+    begin
```

Comment:
do you have thoughts on how we can extend this to also allow override of validation code? I was originally thinking we would have a single route, and only allow levelbuilders to use the override parameters (which might muddy our authorization logic a bit).

Reply:
It didn't end up as reusable as I had hoped for validation, but I'm imagining we write a third `get_project_files` where you can pass in override validation.

I think I prefer the separate routes despite being them being similar, since they have different param requirements and different authorization. Could do some more deduplication probably within the two (maybe 3 when you're done) routes like you suggested though.

Reply:
makes sense